allensdk/brain_observatory/behavior/data_files/_data_file_abc.py

```diff
@@ -0,0 +1,92 @@
+import abc
+from typing import Any, Union
+from pathlib import Path
+
+
+class DataFile(abc.ABC):
+    """An abstract class that prototypes methods for accessing internal
+    data files.
+
+    These data files contain information necessary to sucessfully instantiate
+    one or many `DataObject`(s).
+
+    External users should ignore this class (and subclasses) as as they
+    will only ever be using `from_nwb()` and `to_nwb()` `DataObject` methods.
+    """
+
+    def __init__(self, filepath: Union[str, Path]):  # pragma: no cover
+        self._filepath: str = str(filepath)
+        self._data = self.load_data(filepath=filepath)
```

Comment:
this seems a little "not abstract"
meaning, just looking at a derived class, I can't actually tell where `load_data` gets called.

Reply:
Hmm... the derived classes call `super().__init__()` though which should be enough hint to look at the `DataFile` abc?

The alternative is a lot of repeated code to call `self.load_data`, once for each new `DataFile` subclass

Reply:
I think it is ok. I did figure it out after a few seconds.
I guess I am just not quite sure the distinction between and abstract base class, which I think defines required methods, and a base class, which defines common functionality.
But, I think it is fine.

Reply:
An Abstract Base Class cannot be instantiated

Abstract methods or proerties do not allow subclasses (which inherit from some parent class whose method or methods are decorated with @abstractmethod) to be instantiated if they don't override the abstract method or property